mk-files/semaphore.mk

```diff
@@ -73,5 +73,20 @@ _ci-bin-sem-cache-store:
 .PHONY: ci-bin-sem-cache-restore
 ci-bin-sem-cache-restore:
 	@echo "Restoring semaphore caches"
-	cache restore $(os_name_and_arch)_npm_cache
-	cache restore $(os_name_and_arch)_playwright_cache || true
+	cache restore $(PLATFORM)_$(ARCH)_npm_cache
+	cache restore $(PLATFORM)_$(ARCH)_playwright_cache || true
+
+# Merge per-job blob reports into one HTML report by test job:
+# - Each job's blob reporter will write a report zip to `blob-report/`, so we'll have multiple files
+#  like `blob-report/report-abc123.zip`, `blob-report/report-xyz456.zip` (one for each test)
+# - When we merge them into an HTML report, we'll get one HTML report in `playwright-report/`
+#  containing all tests from each job, which we then zip up and push to Semaphore artifacts
+#  so each job's results can be downloaded separately.
+# (To avoid making large zip files, we're not bundling these into per-agent reports.)
+.PHONY: merge-blob-reports
+merge-blob-reports:
+	npx playwright merge-reports --reporter html blob-report
+	$(eval TEST_SUITE_NAME := $(subst @,, $(or $(TEST_SUITE_TAG),all)))
+	$(eval FILENAME := playwright-report-$(PLATFORM)-$(ARCH)-vscode-$(VSCODE_VERSION)--$(TEST_SUITE_NAME).zip)
+	tar -zcvf $(FILENAME) playwright-report
+	artifact push workflow $(FILENAME) --destination "playwright-reports/$(FILENAME)" --force
```

Comment:
Also, this ends up looking like:

<img width="565" height="604" alt="Image" src="https://github.com/user-attachments/assets/06da33d9-6e79-4772-ad90-8fccdd909267" />